issue_1/test/word_registry/TestWordRegistry.ts

```diff
@@ -0,0 +1,116 @@
+import hre from "hardhat";
+import { Artifact } from "hardhat/types";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/dist/src/signer-with-address";
+
+import { expect } from "chai";
+import { WordRegistry } from "../../typechain/WordRegistry";
+import { Signers } from "../types";
+
+const { deployContract } = hre.waffle;
+
+const NULL_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+describe("Unit tests", function () {
+  before(async function () {
+    this.signers = {} as Signers;
+
+    const signers: SignerWithAddress[] = await hre.ethers.getSigners();
+    this.signers.admin = signers[0];
+    this.signers.user1 = signers[1];
+    this.signers.user2 = signers[2];
+    console.log("admin: ", this.signers.admin.address);
+  });
+
+  describe("WordRegistry", function () {
+    beforeEach(async function () {
+      const artifact: Artifact = await hre.artifacts.readArtifact("WordRegistry");
+      this.wr = <WordRegistry>await deployContract(this.signers.admin, artifact, []);
+    });
+
+    it("should allow users to reserve words", async function () {
+      const word = "sss";
+      await expect(this.wr.connect(this.signers.user1).register(word))
+        .to.emit(this.wr, "OwnershipChange").withArgs(word, NULL_ADDRESS, this.signers.user1.address);
+    });
+
+    it("should preserve ownership of words", async function () {
+      const word = "sss";
+      expect(await this.wr.connect(this.signers.user1).does_own(word))
+        .to.equal(false);
+
+      await this.wr.connect(this.signers.user1).register(word);
+
+      expect(await this.wr.connect(this.signers.user1).does_own(word))
+        .to.equal(true);
+      expect(await this.wr.connect(this.signers.user2).does_own(word))
+        .to.equal(false);
+      expect(await this.wr.connect(this.signers.admin).does_own(word))
+        .to.equal(false);
+    });
+
+    it("should allow users to unregister words", async function () {
+      const word = "sss";
+      await expect(this.wr.connect(this.signers.user1).register(word))
+        .to.emit(this.wr, "OwnershipChange").withArgs(word, NULL_ADDRESS, this.signers.user1.address);
+
+      // user2 can't unregister
+      await expect(this.wr.connect(this.signers.user2).unregister(word))
+        .to.be.reverted;
+
+      await expect(this.wr.connect(this.signers.user1).unregister(word))
+        .to.emit(this.wr, "OwnershipChange").withArgs(word, this.signers.user1.address, NULL_ADDRESS,);
+      expect(await this.wr.connect(this.signers.user1).does_own(word))
+        .to.equal(false);
+    })
+
+    it("should allow words to be re-registered", async function () {
+      const word = "sss";
+      await expect(this.wr.connect(this.signers.user1).register(word))
+        .to.not.be.reverted;
+
+      await expect(this.wr.connect(this.signers.user1).unregister(word))
+        .to.not.be.reverted;
+
+      await expect(this.wr.connect(this.signers.user2).register(word))
+        .to.not.be.reverted;
```

Comment:
No need to check all these, they were tested.

Reply:
80% ack: check on line 71 is redundant, but 74 is legit - it checks that a name can be registered after being unregistered

Reply:
The ones I'm referring to are L69 (tested on L33) and L72 (tested on L58).